src/platforms/alexa/apis/ApiBase.ts

```diff
@@ -0,0 +1,49 @@
+import { RequestEnvelope } from "ask-sdk-model";
+import * as debug from "debug";
+import * as _ from "lodash";
+import * as rp from "request-promise";
+
+const alexalog: debug.IDebugger = debug("voxa:alexa");
+
+export class ApiBase {
+  public errorCodeSafeToIgnore: number = 0; // the code error to ignore on checkError function
+  public tag: string = ""; // the class reference for error logging
+  public voxaEvent: any; // the event as sent by the service
+
+  constructor(event: RequestEnvelope) {
+    this.voxaEvent = _.cloneDeep(event);
+  }
+
+  protected getResult(path = "", method = "GET", body = {}): any {
```

Comment:
the type of each api response can be found in the ask-sdk-model package, for example

services.deviceAddress.Address
services.deviceAddress.ShortAddress

Let's use those, seems like a great oportunitty for [generics](https://www.typescriptlang.org/docs/handbook/generics.html)